app/services/github/contributions/pull_request_collector.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module Github
+  module Contributions
+    class PullRequestCollector
+      LINK_INDEX = {
+        org: 3,
+        repo: 4,
+        pr: 6
+      }
+
+      PR_STATE = {
+        merged: 'merged',
+        open: 'open',
+        closed: 'closed',
+        error: 'Fetch Error'
+      }
+
+      def initialize(client:)
+        @client = client
+      end
+
+      def all
+        valid_pull_requests = Contribution.valid_pull_requests.select(:id, :link, :pr_state)
+        updated_pull_request = valid_pull_requests.map {|contribution| { id: contribution.id, pr_state: fetch_pr_state(contribution.link)} }
+
+        updated_pull_request.select{ |new_pr| validate_update(valid_pull_requests.find_by(id: new_pr[:id]), new_pr) }
```

Comment:
We could use `find` in here instead of `find_by`, right?

Reply:
Ok!